pkg/kubectl/cmd/apply.go

```diff
@@ -373,15 +376,39 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 		}
 		count++
 		if printObject {
-			return cmdutil.PrintObject(cmd, info.Object, out)
+			objs = append(objs, info.Object)
+			return nil
 		}
 		cmdutil.PrintSuccess(shortOutput, out, info.Object, dryRun, "configured")
 		return nil
 	})
-
 	if err != nil {
 		return err
 	}
+
+	if len(objs) > 0 {
```

Comment:
You don't need this check here, below there's `if count == 0` check which returns error. So just move this entire code below it and don't do any other check. Actually, you could drop the `count` variable and test `len(objs)` in that condition.